src/components/CourseFilter/CourseFilter.tsx

```diff
@@ -0,0 +1,192 @@
+  initialTags: any;
+};
+
+export default function CourseFilter({
+  initialCourses,
+  initialTags,
+}: CourseFilterProps) {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+
+  const { palette } = useTheme();
+
+  const [startDate, setStartDate] = useState(null);
+  const [endDate, setEndDate] = useState(null);
+  const [courseName, setCourseName] = useState('');
+  const [selectedTags, setSelectedTags] = useState<string[]>([]);
+
+  const createQueryString = useCallback(
+    (name: string, value: string) => {
+      const params = new URLSearchParams(searchParams);
+      params.set(name, value);
+
+      return params.toString();
+    },
+    [searchParams]
+  );
+
+  useEffect(() => {
+    if (
+      searchParams.get('courseName') === null &&
+      searchParams.get('courseTag') === null &&
+      searchParams.get('courseDates') === null &&
+      searchParams.get('courseId') === null
+    ) {
+      handleClearSearch();
+    }
+  }, [searchParams]);
+
+  const handleNameChange = async (value: string | null) => {
+    const searchTerm: string | null = value;
+    setCourseName(searchTerm || '');
+    router.push(
+      pathname + '?' + createQueryString('courseName', searchTerm || '')
+    );
+  };
+
+  const handleTagChange = async (tagName: string) => {
+    let updatedTags = [];
+    if (selectedTags.includes(tagName)) {
+      updatedTags = selectedTags.filter((tag) => tag !== tagName);
+      setSelectedTags(updatedTags);
+    } else {
+      updatedTags = [...selectedTags, tagName];
+      setSelectedTags(updatedTags);
+    }
+    if (tagName === '') {
+      setSelectedTags([]);
+    }
+    const tagsAsaString = updatedTags.filter((tag) => tag).join(',');
+    router.push(pathname + '?' + createQueryString('courseTag', tagsAsaString));
+  };
+
+  const handleDateChange = async (range: [any, any]) => {
+    const [startDate, endDate] = range;
+    setStartDate(startDate);
+    setEndDate(endDate);
+    if (startDate === null && endDate === null) {
+      router.push(pathname + '?' + createQueryString('courseDates', ''));
+    } else {
+      router.push(
+        pathname +
+          '?' +
+          createQueryString('courseDates', startDate + '-' + endDate)
+      );
+    }
+  };
+
+  const handleClearSearch = async () => {
+    await handleNameChange(null);
+    await handleTagChange('');
+    await handleDateChange([null, null]);
+    router.replace(pathname);
+  };
+
+  const customStyles = {
+    width: '250px',
+    height: '55px',
+    fontSize: '16px',
+  };
+
+  return (
+    <>
+      <Typography variant="h4">Search courses</Typography>
+      <Box
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          padding: '16px',
+          backgroundColor: palette.white.main,
+        }}
+      >
+        <Button
+          variant="contained"
+          onClick={() => handleClearSearch()}
+          sx={{
+            width: '200px',
```

Comment:
Probably no need to hardcode width. You get padding in the MUI Button component anyway.

To save space I would most likely either 1) swap the button and text to an icon with a label (see the alternative layout branch where button group has labels under them or 2) Rename the text to "Clear"